File: samples.cs/Northwind/NorthwindContext.cs
using System;
using System.Collections.Generic;
using System.Data.Linq.Mapping;
using System.Linq;
using System.Text;
using DbExtensions;

namespace Samples.CSharp.Northwind {
   
   public class NorthwindDatabase : Database {

      public SqlTable<Product> Products {
         get { return Table<Product>(); }
      }

      public SqlTable<Order> Orders {
         get { return Table<Order>(); }
      }

      public NorthwindDatabase(string connString, MetaModel mapping) 
         : base(connString, mapping) { }
   }
}

File: samples.vb/Northwind/NorthwindContext.vb
Imports System.Data.Linq.Mapping
Imports DbExtensions

Namespace Northwind

   Public Class NorthwindContext
      Inherits DataAccessObject

      Public Property Products As SqlTable(Of Product)
         Get
            Return Table(Of Product)()
         End Get
         Set(value As SqlTable(Of Product))
         End Set
      End Property

      Public Property Orders As SqlTable(Of Order)
         Get
            Return Table(Of Order)()
         End Get
         Set(value As SqlTable(Of Order))
         End Set
      End Property

      Public Sub New(ByVal connString As String, ByVal mapping As MetaModel)
         MyBase.New(connString, mapping)
      End Sub

   End Class
End Namespace
